src/privileged/trackers/api.js

```diff
@@ -0,0 +1,269 @@
+"use strict";
+
+/* global ExtensionAPI, ExtensionCommon, ExtensionUtils, XPCOMUtils, Services */
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+
+/* eslint-disable-next-line no-var */
+var {EventManager, EventEmitter} = ExtensionCommon;
+/* eslint-disable-next-line no-var */
+var {Management: {global: {tabTracker}}} = ChromeUtils.import("resource://gre/modules/Extension.jsm", {});
+
+XPCOMUtils.defineLazyModuleGetter(
+  this,
+  "BrowserWindowTracker",
+  "resource:///modules/BrowserWindowTracker.jsm",
+);
+
+class TrackersEventEmitter extends EventEmitter {
+  emitReportBreakage(tabId) {
+    this.emit("report-breakage", tabId);
+  }
+  emitReloadWithTrackers(tabId, etld) {
+    this.emit("reload-with-trackers", tabId, etld);
+  }
+  emitPageBeforeUnload(tabId, data) {
+    this.emit("page-before-unload", tabId, data);
+  }
+  emitPageUnload(tabId) {
+    this.emit("page-unload", tabId);
+  }
+  emitErrorDetected(error, tabId) {
+    this.emit("page-error-detected", error, tabId);
+  }
+  emitAddException(tabId) {
+    this.emit("exception-added", tabId);
+  }
+}
+
+/* https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/functions.html */
+this.trackers = class extends ExtensionAPI {
+  constructor(extension) {
```

Comment:
Nice, I didn't think to use the constructor to do setup stuff in my own code for Firefox Monitor because I didn't realize we had access to the extension here already.